extensions/integration-tests/testfmks/src/stress.ts

```diff
@@ -0,0 +1,289 @@
+const trace = require('debug')(`${logPrefix}:trace`);
+/**
+ * Subclass of Error to wrap any Error objects caught during Stress Execution.
+ */
+export class StressError extends Error {
+	inner: Error | any;
+	static code: string = 'ERR_STRESS';
+
+	constructor(error?: any) {
+		super();
+		this.name = StressError.code;
+		this.inner = error;
+		if (error instanceof Error) {
+			this.message = error.message;
+			this.stack = error.stack;
+		} else if (error instanceof String) {
+			this.message = error.valueOf();
+			try {
+				throw new Error();
+			} catch (e) {
+				this.stack = e.stack;
+			}
+		} else if (isString(error)) {
+			this.message = error;
+			try {
+				throw new Error();
+			} catch (e) {
+				this.stack = e.stack;
+			}
+		} else {
+			this.message = 'unknown stress error';
+		}
+	}
+}
+
+/**
+ * Defines an interface to specify the stress options for stress tests.
+ * @param runtime - the number of minutes for which the stress runs. Once this 'runtime' expires stress is terminated even if we have not exceeded {@link iterations} count yet. NYI. This is here for future use to allow really long running stress tests. Default value is provided by environment variable: StressRuntime and if undefined then by {@link DefaultStressOptions}.
+ * @param dop - the number of parallel instances of the decorated method to run. Default value is provided by environment variable: StressDop and if undefined then by {@link DefaultStressOptions}.
+ * @param iterations - the number of iterations to run in each parallel invocation for the decorated method. {@link runtime} can limit the number of iterations actually run. Default value is provided by environment variable: StressIterations and if undefined then by {@link DefaultStressOptions}.
+ * @param passThreshold - the fractional number of all invocations of the decorated method that must pass to declared the stress invocation of that method to be declared passed. Range: 0.0-1.0. Default value is provided by environment variable: StressPassThreshold and if undefined then by {@link DefaultStressOptions}.
+ */
+export interface StressOptions {
+	runtime?: number;
+	dop?: number;
+	iterations?: number;
+	passThreshold?: number;
+}
+
+/**
+ * The default values for StressOptions.
+ */
+export const DefaultStressOptions: StressOptions = { runtime: 120, dop: 4, iterations: 10, passThreshold: 0.95 };
+
+/**
+ * Defines the shape of stress result object
+ */
+export interface StressResult {
+	numPasses: number;
+	fails: Error[];
+	errors: Error[];
+}
+
+
+/**
+* This simulates a sleep where the thread is suspended without spinning for a given number of milliseconds before resuming
+*/
+export async function sleep(ms: number) {
+	return await (async () => {
+		return await new Promise((undefined) => setTimeout(undefined, ms));
+	})();
+}
+
+/**
+* This is just a synonym for sleep(0). This has the affect of yielding to other operations.
+*/
+export async function bear() {
```

Comment:
any reason why this is necessary then?

Reply:
The name makes its intention clearer. The purpose of calling sleep(0) is to give the execution time slice to other running threads.

Reply:
Ideal name would have been yield but that is a reserved keyword and 'bear' is a synonym of sleep.

Reply:
it is there because it is a very important thing to do (that is making a call with sleep(0), so while not necessary, it presence and name documents whey we need to make sleep(0) call.

---
In reply to: [284923292](https://github.com/microsoft/azuredatastudio/pull/5499#discussion_r284923292) [](ancestors = 284923292)